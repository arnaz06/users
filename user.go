package users

import (
	"context"
	"time"
)

// User is the struct represent the user's data
type User struct {
	ID          string    `json:"id"`
	Email       string    `json:"email" validate:"required"`
	Address     string    `json:"address"`
	Password    string    `json:"password" validate:"required"`
	CreatedTime time.Time `json:"created_time"`
	UpdatedTime time.Time `json:"updated_time"`
}

// UserRepository is interface of user repository.
type UserRepository interface {
	Create(ctx context.Context, user User) (User, error)
	Get(ctx context.Context, id string) (User, error)
	GetByEmail(ctx context.Context, email string) (User, error)
	Update(ctx context.Context, user User) error
	Delete(ctx context.Context, id string) error
}

// UserService is interface of user service.
type UserService interface {
	Create(ctx context.Context, user User) (User, error)
	Get(ctx context.Context, id string) (User, error)
	Login(ctx context.Context, email, password string)  error
	Update(ctx context.Context, user User) error
	Delete(ctx context.Context, id string) error
}
