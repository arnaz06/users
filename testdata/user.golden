{
    "id": "123",
    "email": "jhon@doe.com",
    "password": "secret-123",
    "address": "lorem ipsum lorem ipsum",
    "created_time": "2020-08-29T09:32:25+07:00",
    "updated_time": "2020-08-29T09:32:25+07:00"
}
